askama_shared/src/generator.rs

```diff
@@ -1135,6 +1138,39 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
         })
     }
 
+    fn _visit_escape_filter(
+        &mut self,
+        buf: &mut Buffer,
+        args: &[Expr<'_>],
+    ) -> Result<(), CompileError> {
+        if args.len() > 2 {
+            return Err("only two arguments allowed to escape filter".into());
+        }
+        let opt_escaper = match args.get(1) {
+            Some(Expr::StrLit(name)) => Some(*name),
+            Some(_) => return Err("invalid escaper type for escape filter".into()),
+            None => None,
+        };
+        let escaper = match opt_escaper {
+            Some(name) => self
+                .input
+                .config
+                .escapers
+                .iter()
+                .filter(|(escapers, _)| escapers.contains(name))
```

Comment:
Instead of `.filter().map().next()` can we just use `find()`?

Reply:
Yeah sure, using `.find()` reduces it by a line :)

```rust
let escaper = match opt_escaper {
    Some(name) => self
        .input
        .config
        .escapers
        .iter()
        .find(|(escapers, _)| escapers.contains(name))
        .map(|(_, escaper)| escaper)
        .ok_or(CompileError::Static("invalid escaper for escape filter"))?,
    None => self.input.escaper,
};
```

Actually, what version are we targetting? If we allow Rust 1.50.0, then we can use [`.then()`](https://doc.rust-lang.org/std/primitive.bool.html#method.then) and reduce it with an additional line more.

```rust
let escaper = match opt_escaper {
    Some(name) => self
        .input
        .config
        .escapers
        .iter()
        .find_map(|(escapers, escaper)| escapers.contains(name).then(|| escaper))
        .ok_or(CompileError::Static("invalid escaper for escape filter"))?,
    None => self.input.escaper,
};
```

Reply:
Yeah, requiring 1.50.0 for `bool::then()` sounds good.